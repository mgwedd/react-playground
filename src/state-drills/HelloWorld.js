import React, { Component } from 'react';

export default class HelloWorld extends Component {
    
    state = {
        who: 'World'
    }

    handleButtonClick = (e) => {
        this.setState({who: e.target.value});
        console.log('this is state: ', this.state)
    }
    render() {
        return (
            <div>
                <p>Hello, {this.state.who}.</p>
                <button onClick={(e) => this.handleButtonClick(e)} value='World'>World</button>
                <button onClick={(e) => this.handleButtonClick(e)} value='Friend'>Friend</button>
                <button onClick={(e) => this.handleButtonClick(e)} value='React'>React</button>
            </div>
        );
    }
}