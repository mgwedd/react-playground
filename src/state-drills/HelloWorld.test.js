import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import HelloWorld from './HelloWorld';

describe('HelloWorld Component', () => {

    it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<HelloWorld />, div);
    ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
    const tree = renderer
        .create(<HelloWorld />)
        .toJSON();
    expect(tree).toMatchSnapshot();  
    });
    
});