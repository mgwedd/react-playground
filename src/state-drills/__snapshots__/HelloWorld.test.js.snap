// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld Component renders the UI as expected 1`] = `
<div>
  <p>
    Hello, 
    World
    .
  </p>
  <button
    onClick={[Function]}
    value="World"
  >
    World
  </button>
  <button
    onClick={[Function]}
    value="Friend"
  >
    Friend
  </button>
  <button
    onClick={[Function]}
    value="React"
  >
    React
  </button>
</div>
`;
